File: translations/codenet/deepseek_commented/Go2C++/codellama/Code/s860368733.cpp

#include <iostream>
#include <map>

using namespace std;

int main() {
	int n, m; // Declare variables to store the number of elements and comparisons
	cin >> n >> m; // Read the number of elements and comparisons from standard input
	int h[n]; // Create an array to store the heights of the elements
	for (int i = 0; i < n; i++) {
		cin >> h[i]; // Read the heights of the elements from standard input
	}
	int a, b; // Declare variables to store the indices of the elements being compared
	map<int, int> counter; // Create a map to keep track of elements that cannot be the highest
	for (int i = 0; i < m; i++) {
		cin >> a >> b; // Read each comparison from standard input
		if (h[a-1] < h[b-1]) { // Check if the height of the first element is less than the second
			counter[a] = 1; // Mark the first element as not being the highest
		} else if (h[a-1] > h[b-1]) { // Check if the height of the first element is greater than the second
			counter[b] = 1; // Mark the second element as not being the highest
		} else if (h[a-1] == h[b-1]) { // Check if the heights of the two elements are equal
			counter[a] = 1; // Mark both elements as not being the highest
			counter[b] = 1;
		}
	}
	cout << n - counter.size() << endl; // Print the number of elements that cannot be the highest
	return 0;
}

File: translations/codenet/deepseek_commented/Go2P/deepseek/Code/s860368733.py
def main():
    n, m = map(int, input().split())  # Read the number of elements and comparisons
    h = list(map(int, input().split()))  # Read the heights of the elements
    counter = set()  # Create a set to keep track of elements that cannot be the highest
    for _ in range(m):
        a, b = map(int, input().split())  # Read each comparison
        if h[a - 1] < h[b - 1]:  # Check if the height of the first element is less than the second
            counter.add(a)  # Mark the first element as not being the highest
        elif h[a - 1] > h[b - 1]:  # Check if the height of the first element is greater than the second
            counter.add(b)  # Mark the second element as not being the highest
        else:  # If the heights of the two elements are equal
            counter.add(a)  # Mark both elements as not being the highest
            counter.add(b)
    print(n - len(counter))  # Print the number of elements that cannot be the highest


if __name__ == "__main__":
    main()
